Browse next upgrades

hardiesoft opened the pull request.

CameronRP commented:
Currently the only other state that shows up is `Tomp4`. We will have a few more states when we add audio recordings to the recordings table.
I don't think we are needing to re-order the columns at the moment. But after combining the audio and thermal recordings we will probably make a few changes then.

hardiesoft committed:
- Updated bootstrap-vue-next lib
- Updated Vue3
- Reworking mask regions component to disallow self-intersecting regions, and allow an apron so regions can be created slightly outside the camera frame.
- Creating wizard steps for manual uploads
- Moving device status out of modal

hardiesoft committed:
- Fix mask regions tests

hardiesoft committed:
- Fix mask regions tests (2, linting)

hardiesoft committed:
- Fix sunset/sunrise in visits breakdown
- Add more info to mobile device cards in DevicesView

hardiesoft committed:
- Fix reference photo component, make it work on mobile.
- More responsive layouts for device setup.

hardiesoft committed:
- Reference photos are properly scaled when overlaying on CptvPlayer on mobile.
- Overflowing tab list component to make longer tabbed nav lists work better on mobile.

hardiesoft committed:
- Trail cameras use an appropriate daytime image as an automatic reference image.
- Masked regions are now used when adding tracks via the API to ignore tracks that are exclusively inside the masked region.
- Use common track rect drawing code for showing tracks for trailcam images.

hardiesoft committed:
- Added a dummy track to trailcam recordings where there is no track – the actual track is created the first time a user classification is added.  This could be extended to empty CPTV files.

hardiesoft committed:
- Styling dummy track

hardiesoft committed:
- Fix tests

hardiesoft committed:
- Added ability to add a dummy track to a CPTV file that has no tracks.
- Automatically add "missed track" tag to CPTV recordings that get manual IDs when they had no tracks.
- Poll every 30 seconds when an open recording is still processing, and update UI when processing is complete.

hardiesoft committed:
- Improved behaviour when mashing on next/prev buttons in CPTV player very quickly :)

hardiesoft committed:
- Fix null settings read in mask-regions API

hardiesoft committed:
- Remove stations wording on Dashboard

hardiesoft committed:
- API Documentation fix
- Add "dummy-track" to tracker version to make it available to processing.

hardiesoft committed:
- Don't send algorithm with tracks, use default

hardiesoft committed:
- Fix mask-regions api read

hardiesoft committed:
- Clone rather than always update in place DeviceHistory entries for adding reference images and mask regions in the same location.
- Fix route for Visits modal from activity view to include selected track.

hardiesoft committed:
- Clone rather than always update in place DeviceHistory entries for adding reference images and mask regions in the same location.
- Fix route for Visits modal from activity view to include selected track. (2)

hardiesoft committed:
- Improved responsive layout for activity search
- Make search url query driven (WIP)

hardiesoft committed:
- Improved responsive layout for activity search
- Make search url query driven (WIP)
- Fix mask-regions api

hardiesoft committed:
- Improved responsive layout for activity search
- Make search url query driven
- Hack in support for "Animal" class from Megadetector
- Show processing progress for visits in API
- Reflect processing progress in the UI.

hardiesoft committed:
- Add UI to visits when recordings are still processing
- Handle scaling DPI when drawing track boxes on trailcam images

hardiesoft committed:
- Make sure player inherits query params from parent component
- Make recording/visits exports work.

hardiesoft committed:
- Don't delete previous reference images

hardiesoft committed:
Scale UP:
- New bulk querying/reporting API which is *many*, *many* times faster for tag filtering queries over longer periods, esp. when the number of recordings in a project grows.
- Thumbnail images on the front-end only get queried when visible.

hardiesoft committed:
Scale UP:
- CI Tests for new bulk tag mode.

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-upgrades

# Conflicts:
#	api/api/V1/Device.ts
#	api/api/V1/Recording.ts
#	api/api/V1/recordingUtil.ts
#	api/models/index.ts
#	api/package-lock.json
#	browse-next/src/api/Recording.ts
#	browse-next/src/components/ActivitySearchDescription.vue
#	browse-next/src/components/ImageLoader.vue
#	browse-next/src/components/RecordingsList.vue
#	browse-next/src/components/VisitsDailyBreakdown.vue
#	browse-next/src/views/ActivitySearchView.vue

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-upgrades

hardiesoft committed:
Merge main

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-upgrades

hardiesoft committed:
Merge main, add all tagModes to improved bulk query.

Zainrax commented on api/api/V1/Device.ts:
```diff
@@ -1393,7 +1393,7 @@ export default function (app: Application, baseUrl: string) {
   app.get(
     `${apiUrl}/:id/settings`,
-    extractJwtAuthorizedUserOrDevice,
+    extractJwtAuthorizedUser,
```
We probably want to allow devices to use this endpoint as well, though it may not be setup properly for devices due to fetchAuthorizedRequiredDeviceById being present(?).

hardiesoft committed:
Address review feedback.

hardiesoft committed:
Small fixes

hardiesoft committed:
Small fixes (2)

hardiesoft committed:
Small fixes (3)

hardiesoft committed:
Small fixes (3)

hardiesoft committed:
Small fixes (4)

hardiesoft committed:
Get rid of legacy bulk recordings query in browse-next
